### PR TITLE
Add blocknote

### PR DESCRIPTION
**Pull request by CTNicholas**
This adds [BlockNote](https://www.blocknotejs.org/). 

- Getting started guide
- Basic example
- Advanced example
- https://github.com/liveblocks/liveblocks.io/pull/1872

image

### CODE REVIEW

File: docs/pages/get-started/yjs-blocknote-react.mdx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+        if (!doc || !provider) {
+          return null;
+        }
+
+        return <BlockNoteEditor doc={doc} provider={provider} />;
+      }
+
+
+      function BlockNoteEditor({ doc, provider }: EditorProps) {
```
Comment (YousefED) [inline docs/pages/get-started/yjs-blocknote-react.mdx:137]: this conflicts with the imported `BlockNoteEditor`

File: docs/pages/get-started/yjs-blocknote-react.mdx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  title: "Get started with Liveblocks, Yjs, BlockNote, and React"
+  parentTitle: "Get started"
+  description:
+    "Learn how to get started with Liveblocks, Yjs, BlockNote, and React."
+---
+
+Liveblocks is a real-time collaboration infrastructure for building performant
+collaborative experiences. Follow the following steps to start adding
+collaboration to your React application using the APIs from the
+[`@liveblocks/yjs`](/docs/api-reference/liveblocks-yjs) package.
```
Comment (YousefED) [inline docs/pages/get-started/yjs-blocknote-react.mdx:12]: Maybe add something like:

```
[BlockNote](https://www.blocknotejs.org) is an open source Block-Based rich text editor that comes with a Notion-style UI. BlockNote supports using Yjs and LiveBlocks for real time collaboration.
```

File: docs/pages/get-started/yjs-blocknote-react.mdx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  <Step>
+    <StepTitle>Join a Liveblocks room</StepTitle>
+    <StepContent>
+
+      Liveblocks uses the concept of rooms, separate virtual spaces where people
+      collaborate. To create a real-time experience, multiple users must be
+      connected to the same room.
+
+      ```tsx file="index.tsx" highlight="7-11"
+      import { RoomProvider } from "./liveblocks.config";
```
Comment (YousefED) [inline docs/pages/get-started/yjs-blocknote-react.mdx:76]: perhaps add `"use client";` for nextjs compatibility. Same for Editor.tsx

**Review by YousefED — COMMENTED** [review]
Works seamlessly, great stuff! Left some fixes in the review.

One other thing I struggle with nowadays is the "initial step" for explaining React constructs. I.e.; what is missing, is how I scaffold a new project (the tutorial assumes you already have a React app running).

If you don't have a React app running, the official docs @ https://react.dev/learn/start-a-new-react-project nowadays suggest to use `npx create-next-app` and thus Next.js to setup your project. Unfortunately (imo), this then makes the tutorial a bit incompatible, as Next.js requires specific file names (e.g.: move `index.tsx` to `./app/demo/page.tsx` or similar).

What do you think about this, is this something you run into when explaining liveblocks to devs?

(least we can do is add `"use client";` btw, as suggested in the comments)

**Comment by CTNicholas** [comment]
> Can someone from the Liveblocks team review it (fix the change requests and merge it)? Maybe @CTNicholas can you please help me? I need this to my new project!

Hiya! I'm planning on working on this when I can make examples too 🙂 This is just a getting started guide, so Liveblocks Yjs already works with BlockNote. Just follow the guide here:

https://github.com/liveblocks/liveblocks/blob/bc9f9b9c5ff0d601d37c2cec5730225f88957b22/docs/pages/get-started/yjs-blocknote-react.mdx

**Comment by Gyurmatag** [comment]
> > Can someone from the Liveblocks team review it (fix the change requests and merge it)? Maybe @CTNicholas can you please help me? I need this to my new project!
> 
> Hiya! I'm planning on working on this when I can make examples too 🙂 This is just a getting started guide, so Liveblocks Yjs already works with BlockNote. Just follow the guide here:
> 
> https://github.com/liveblocks/liveblocks/blob/bc9f9b9c5ff0d601d37c2cec5730225f88957b22/docs/pages/get-started/yjs-blocknote-react.mdx

Thanks! Will it work with Next.js serverless functions on Vercel. Am I right?

**Comment by CTNicholas** [comment]
> > > Can someone from the Liveblocks team review it (fix the change requests and merge it)? Maybe @CTNicholas can you please help me? I need this to my new project!
> > 
> > 
> > Hiya! I'm planning on working on this when I can make examples too 🙂 This is just a getting started guide, so Liveblocks Yjs already works with BlockNote. Just follow the guide here:
> > https://github.com/liveblocks/liveblocks/blob/bc9f9b9c5ff0d601d37c2cec5730225f88957b22/docs/pages/get-started/yjs-blocknote-react.mdx
> 
> Thanks! It will work with Next.js serverless functions on Vercel. Am I right?

Yes! Like our [other examples](https://liveblocks.io/examples/browse/text-editor).

**Comment by Gyurmatag** [comment]
Hey @CTNicholas on the example I've got an error when I move my mouse over the editor. Here it is:

Képernyőfotó 2023-11-10 - 21 33 17

What could be the problem here?

**Comment by YousefED** [comment]
> Hey @CTNicholas on the example I've got an error when I move my mouse over the editor. Here it is:
> 
> Képernyőfotó 2023-11-10 - 21 33 17
> 
> What could be the problem here?

Hi! I'm fairly sure this should be fixed when you upgrade to the latest version of BlockNote. If not, feel free to open an issue on our repo.

@CTNicholas what do you need to get this PR merged? lmk if I can help!

**Comment by CTNicholas** [comment]
> @CTNicholas what do you need to get this PR merged? lmk if I can help!

Hey, sorry it's been a little slow! I'd like to create some proper examples and guides too, which is why this didn't get merged. I'll be looking into it again soon!

File: examples/nextjs-yjs-blocknote-advanced/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  </a>
+</p>
+
+# Advanced Collaborative Rich Text Editor (BlockNote)
+
+<p>
+  <a href="https://liveblocks.io/examples/collaborative-text-editor-advanced/nextjs-yjs-blocknote-advanced/preview">
+    <img src="https://img.shields.io/badge/live%20preview-message?style=flat&logo=data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjQgMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PHBhdGggZD0iTTE2Ljg0OSA0Ljc1SDBsNC44NDggNS4wNzV2Ny4wMDhsMTItMTIuMDgzWk03LjE1IDE5LjI1SDI0bC00Ljg0OS01LjA3NVY3LjE2N2wtMTIgMTIuMDgzWiIgZmlsbD0iI2ZmZiIvPjwvc3ZnPg==&color=333" alt="Live Preview" />
+  </a>
+  <a href="https://codesandbox.io/s/github/liveblocks/liveblocks/tree/main/examples/nextjs-yjs-blocknote">
```
Comment (marcbouchenoire) [inline examples/nextjs-yjs-blocknote-advanced/README.md:16]:
```suggestion
  <a href="https://codesandbox.io/s/github/liveblocks/liveblocks/tree/main/examples/nextjs-yjs-blocknote-advanced">
```

**Review by marcbouchenoire — APPROVED** [review]
LGTM, without looking too deep at Yjs/BlockNote specifics though.

**Comment by YousefED** [comment]
Nice! I'll have a look later today / tomorrow

**Comment by CTNicholas** [comment]
> Nice! I'll have a look later today / tomorrow

Thanks for the feedback before 🙂 I made changes you suggested, and you don't need to check over this PR, I'm quite happy with it! With this PR, and another hidden PR, we're adding:

- BlockNote to our [get started pages](https://liveblocks.io/docs/get-started/text-editor)
- Basic BlockNote and advanced BlockNote examples, which will be [deployed and visible here](https://liveblocks.io/examples/browse/text-editor)
- Both examples will also be downloadable with our CLI, for example:
```shell
npx create-liveblocks-app@latest --example nextjs-yjs-blocknote-advanced
```

We'd also love to promote BlockNote on socials, as it's an awesome framework, and we're planning to do this this week! Do you think we could add Liveblocks to the BlockNote docs too? I've made a PR for it here, and it should be good to merge in a couple of days:
- https://github.com/TypeCellOS/BlockNote/pull/540

**Comment by CTNicholas** [comment]
I'm merging this now so I can deploy the new examples and get our website PR done 👍

File: examples/nextjs-yjs-blocknote-advanced/README.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  </a>
+  <img src="https://img.shields.io/badge/react-message?style=flat&logo=react&color=0bd&logoColor=fff" alt="React" />
+  <img src="https://img.shields.io/badge/next.js-message?style=flat&logo=next.js&color=07f&logoColor=fff" alt="Next.js" />
+  <img src="https://img.shields.io/badge/yjs-message?style=flat&color=0bd" alt="Yjs" />
+  <img src="https://img.shields.io/badge/blocknote-message?style=flat&color=07f" alt="BlockNote" />
+</p>
+
+This example shows how to build a collaborative text editor with
+[Liveblocks](https://liveblocks.io), [Yjs](https://docs.yjs.dev),
+[BlockNote](https://blocknote.dev), and [Next.js](https://nextjs.org/).
```
Comment (YousefED) [inline examples/nextjs-yjs-blocknote-advanced/README.md:27]: @CTNicholas this should point to https://www.blocknotejs.org (I think this file is only used a README on git right?)

**Review by YousefED — COMMENTED** [review]
This is amazing, your docs website + system is really something to be jealous of ❤️

**Comment by CTNicholas** [comment]
Thanks for the feedback @YousefED! Your recommendations are merged:
- https://github.com/liveblocks/liveblocks/pull/1425

**Pull request merged**